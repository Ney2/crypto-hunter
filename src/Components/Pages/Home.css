* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.header-container {
  display: flex;
  justify-content: space-between;
  padding: 2%;
}

.accessories {
  display: flex;
  gap: 15px;
}

.crypto-img {
  position: relative;
  text-align: center;
  font-size: 20px;
  font-weight: bolder;
}

.crypto-img img {
  width: 100vw;
}

.filter {
  position: absolute;
  top: 50%;
  left: 50%;
  background: #fff;
  border: none;
  width: 50%;
  height: 7%;
  border-radius: 30px;
  padding: 2%;
  transform: translate(-50%, -50%);
}

.back-arrow {
  position: absolute;
  top: 5%;
  left: 5%;
  background-color: #fff;
}

.coin-list {
  display: grid;
  grid-template-columns: repeat(3, 33%);
}

.coin {
  border: 2px solid black;
  padding: 2%;
}

.first-container {
  display: flex;
  justify-content: space-between;
  padding: 2%;
}

.pic {
  width: 50%;
  height: 50%;
}

.second-container {
  text-align: end;
  padding: 2%;
}
