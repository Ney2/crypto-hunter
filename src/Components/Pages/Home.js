import React from 'react';
import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import * as BsIcon from 'react-icons/bs';
import './Home.css';

const Home = ({
  name, pic, price, id,
}) => {
  const navigate = useNavigate();
  const navigateDetails = () => {
    navigate(`/details/${id}`);
  };

  return (
    <section>
      <div className="coin">
        <div className="first-container">
          <img className="pic" src={pic} alt={name} />
          <BsIcon.BsFillArrowRightCircleFill className="arrow" id={id} onClick={navigateDetails} key={id} />
        </div>
        <div className="second-container">
          <h2 className="name">{name}</h2>
          <p className="price">{price}</p>
        </div>
      </div>
    </section>
  );
};

Home.propTypes = {
  name: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  id: PropTypes.string.isRequired,
  pic: PropTypes.node.isRequired,
};

export default Home;
