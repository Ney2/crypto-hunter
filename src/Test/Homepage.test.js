import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/configureStore';
import Home from '../Components/Pages/Home';

test('renders correctly', () => {
  const tree = Renderer.create(
    <React.StrictMode>
      <Provider store={store}>
        <BrowserRouter>
          <Home />
        </BrowserRouter>
      </Provider>
    </React.StrictMode>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
