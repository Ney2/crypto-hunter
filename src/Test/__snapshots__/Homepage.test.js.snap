// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section>
  <div
    className="coin"
  >
    <div
      className="first-container"
    >
      <img
        className="pic"
      />
      <svg
        className="arrow"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
        />
      </svg>
    </div>
    <div
      className="second-container"
    >
      <h2
        className="name"
      />
      <p
        className="price"
      />
    </div>
  </div>
</section>
`;
